### PR TITLE
Fix text_input_v3 preedit string

### PR DESCRIPTION
**Pull request by wengxt**
## Description

For every batch of text_input_v3 updates, if there is no preedit in this
batch, preedit should be cleared.

### CODE REVIEW

**Comment by flibitijibibo** [comment]
Makes sense to me... out of curiosity what's a good way to test this, I had a lot of difficulty testing certain text input things because I have close to zero experience past ASCII input.

**Comment by wengxt** [comment]
> Makes sense to me... out of curiosity what's a good way to test this, I had a lot of difficulty testing certain text input things because I have close to zero experience past ASCII input.

In KDE, it will just start an in process compositor, but kwin itself has more library to use... Weston doesn't have text input v3 support so there's no luck here. Not sure if using wlroots can be easier for SDL project.

File: src/video/wayland/SDL_waylandevents.c [hunk 1]
```diff
@@ -1557,7 +1559,12 @@ text_input_done(void *data,
                 struct zwp_text_input_v3 *zwp_text_input_v3,
                 uint32_t serial)
 {
-    /* No-op */
+    SDL_WaylandTextInput *text_input = data;
+    if (!text_input->has_preedit) {
+        char buf[1] = {'\0'};
+        SDL_SendEditingText(buf, 0, 0);
```
Comment (flibitijibibo) [inline src/video/wayland/SDL_waylandevents.c:1565]: Looked around the rest of the source, turns out other backends do it like this:

```suggestion
        SDL_SendEditingText("", 0, 0);
```

**Review by flibitijibibo — CHANGES_REQUESTED** [review]
Looked at the other backends for reference and they pretty much do what this does, so with this one change we're good to go.

**Review by flibitijibibo — APPROVED** [review]
Perfect, thanks for checking this out!

**Pull request merged**